File: docs/docs/src/node/user-guide/io/csv.js
// --8<-- [start:setup]
const pl = require("nodejs-polars");
// --8<-- [end:setup]

`
// --8<-- [start:read]
df = pl.readCSV("path.csv")
// --8<-- [end:read]
`;

// --8<-- [start:write]
df = pl.DataFrame({ foo: [1, 2, 3], bar: [null, "bak", "baz"] });
df.writeCSV("path.csv");
// --8<-- [end:write]

// --8<-- [start:scan]
df = pl.scanCSV("path.csv");
// --8<-- [end:scan]

File: docs/docs/src/rust/user-guide/io/csv.rs
use polars::prelude::*;

fn main() -> Result<(), Box<dyn std::error::Error>>{

    """
    // --8<-- [start:read]
    use polars::prelude::*;

    let df = CsvReader::from_path("path.csv").unwrap().finish().unwrap();
    // --8<-- [end:read]
    """

    // --8<-- [start:write]
    let mut df = df!(
        "foo" => &[1, 2, 3],
        "bar" => &[None, Some("bak"), Some("baz")],
    )
    .unwrap();
    
    let mut file = std::fs::File::create("path.csv").unwrap();
    CsvWriter::new(&mut file).finish(&mut df).unwrap();
    // --8<-- [end:write]

    // --8<-- [start:scan]
    let df = LazyCsvReader::new("./test.csv").finish().unwrap();
    // --8<-- [end:scan]

    Ok(())
}